File: 2022/day06/javascript/part1.mjs
import { join } from "node:path";
import { readFile } from "node:fs/promises";

const filePath = new URL(join(import.meta.url, "..", "..", "input.txt"));

const file = await readFile(filePath, { encoding: "utf-8" });
const data = file.split("\n")[0];

const memory = [];
let i = 0;
for (; i < data.length; i++) {
  const letter = data[i];
  memory.splice(0, memory.indexOf(letter) + 1);
  memory.push(letter);

  if (memory.length === 4) break;
}

console.log(i + 1);

File: 2022/day06/python/part1.py
from pathlib import Path
file_path = Path(__file__, '..', '..', 'input.txt').resolve()

with open(file_path) as f:
    data = f.read().splitlines()[0]
    memory = []

    for i, letter in enumerate(data):
        if letter in memory:
            del memory[:memory.index(letter)+1]
        memory.append(letter)

        if (len(memory) == 4):
            break

    print(i+1)
